cpp/src/slice2rb/RubyUtil.cpp

```diff
@@ -583,7 +582,7 @@ Slice::Ruby::CodeVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
                 _out << "::Ice::OperationMode::Idempotent";
                 break;
         }
-        _out << ", " << ((p->hasMetadata("amd") || (*s)->hasMetadata("amd")) ? "true" : "false") << ", " << format
+        _out << ", " << ((p->hasMetadata("amd") || op->hasMetadata("amd")) ? "true" : "false") << ", " << format
```

Comment:
I wonder why Ruby has this "amd" checks, when there isn't a server side mapping, looks suspicious.

Reply:
Would be good to open an issue to review this and probably remove it.